src/test/scala/com/actian/spark_vector/datastream/DataStreamPartitionAssignmentTest.scala

```diff
@@ -56,6 +57,15 @@ class DataStreamPartitionAssignmentTest extends FunSuite with Matchers {
     /** make partitions have affinity to nodes that are not necessarily VectorEndpoints */
     edges <- Gen.listOfN(nA, Gen.resize(replFactor, Gen.listOf(Gen.choose(0, nB * 2 - 1).map(b => s"host-$b"))))
   } yield (edges, generateEndPoints(nB))
+  
+  private val unevenEndpointsPerHostGen = for {
+    nB <- Gen.choose(3, 20)
+    nA <- Gen.choose(0, 1200)
+    /** all partitions are replicated on all nodes */
+    edges = (0 until nA).map(a => (0 until nB).map(b => s"host-$b"))
```

Comment:
what is `a`'s role in here? 

just out of curiosity, is it me or you're creating nA x "host-$b" foreach $b in nB strings here? how you're identifying the left node from the bigraph then?